cpp/arcticdb/entity/data_error.hpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#pragma once
+
+#include <string>
+
+#include <fmt/format.h>
+
+#include <arcticdb/pipeline/query.hpp>
+
+namespace arcticdb::entity {
+
+enum class VersionRequestType: uint32_t {
+    SNAPSHOT,
```

Comment:
This is quite similar to the LoadType enum, do they need to be kept in sync with each other for ever? Might it be worth having a function that converts from one to the other, i.e. from the thing we tried to the class of operation that failed?

Reply:
I was a bit concerned about this as well. Not a blocker but good if we can keep them in sync somehow.

Reply:
It looks more like the `VersionQueryType` I added. It would be better if all those classes were exposed to Python, instead of the `set_*` methods on `VersionQuery` right now, and we could just return one of those combining the `version_request_type` and `version_request_data` return parts right now.